tianshou/data/batch.py

```diff
@@ -45,9 +45,13 @@ def _valid_bounds(length: int, index: Union[
 
 def _create_value(inst: Any, size: int) -> Union['Batch', np.ndarray]:
     if isinstance(inst, np.ndarray):
+        if issubclass(inst.dtype.type, (np.bool_, np.number)):
```

Comment:
Are you sure `isinstance(inst.dtype, (np.bool_, np.number))` is not doing the same ? I'm interested to see a case were it is not, since `isinstance` is also checking inheritance.

Reply:
```
a = np.asarray([False, False])
isinstance(a.dtype, np.bool_) # False
```

``inst.dtype`` is actually of type ``np.dtype``.

Reply:
Ah yes I see, so it is the same as `isinstance(inst.item(0), (np.bool_, np.number))` but `inst.item(0)` is doing an extra copy so I guess it is slower.

maybe you could have a look to `np.can_cast` instead of `isinstance`:
```
np.can_cast(np.asarray([1.0, 1.0]), np.inexact)
>>> True
np.can_cast(np.asarray([1.0, 1.0]), np.bool_')
>>> False
np.can_cast(np.asarray([False, False]), np.bool_)
>>> True
```

Reply:
``np. can_cast `` and ``np.issubdtype`` both works, but they cannot test multiple types in one function call. I choose ``issubclass`` because it supports multiple tests at the same time by calling ``issubclass(a, (b, c, d))``.

Reply:
Indeed.